src/ServiceControl.Plugin.Nsb6.SagaAudit/AuditInvokedSagaBehavior.cs

```diff
@@ -18,22 +20,27 @@ class AuditInvokedSagaBehavior : Behavior<IInvokeHandlerContext>
                 return;
             }
 
-            if (activeSagaInstance.Instance.Entity == null)
+            AddOrUpdateInvokedSagasHeader(context.Headers, activeSagaInstance.Instance);
+        }
+
+        public static void AddOrUpdateInvokedSagasHeader(Dictionary<string, string> headers, Saga sagaInstance)
+        {
+            if (sagaInstance.Entity == null)
             {
-                return; // Message was not handled by the saga
+                return;
             }
 
-            var invokedSagaAuditData = $"{activeSagaInstance.Instance.GetType().FullName}:{activeSagaInstance.Instance.Entity.Id}";
+            var invokedSagaAuditData = $"{sagaInstance.GetType().FullName}:{sagaInstance.Entity.Id}";
 
             string invokedSagasHeader;
 
-            if (context.MessageHeaders.TryGetValue(SagaAuditHeaders.InvokedSagas, out invokedSagasHeader))
+            if (headers.TryGetValue(SagaAuditHeaders.InvokedSagas, out invokedSagasHeader))
             {
-                context.Headers[SagaAuditHeaders.InvokedSagas] += $"{invokedSagasHeader};{invokedSagaAuditData}";
+                headers[SagaAuditHeaders.InvokedSagas] = $"{invokedSagasHeader};{invokedSagaAuditData}";
```

Comment:
The current version of SC blows up even before it gets to this header. It first tries to interpret the header containing state changes and throws an exception. In theory, if it did not throw, it would process the correct InvokedSagas header properly.